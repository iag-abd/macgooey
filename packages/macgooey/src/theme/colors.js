// @flow

import { type ColorPalette, type ColorShade, type Styles } from "../theme";

type ThemeUtil = {|
  theme: Styles
|};

type ColorUtil = {
  ...ThemeUtil,
  color: string
};

// TODO remove the `Object` type declarations
export const themeColor = (color: $Keys<ColorPalette> = "primary") => ({
  theme
}: ThemeUtil) => theme.color[color];

export const themeColorCompliment = (
  compliment: $Keys<ColorPalette> = "primary"
) => ({ theme }: ThemeUtil) => theme.colorCompliment[compliment];

export const themeColorShade = (shade: $Keys<ColorShade> = "gray") => ({
  theme
}: ThemeUtil) => theme.colorShade[shade];

// If the element takes a `color` prop
export const colorOr = (defaultColor: string = "#000") => ({
  color,
  theme
}: ColorUtil) => theme.color[color] || defaultColor;

export const complimentOr = (defaultColor: string = "#fff") => ({
  color,
  theme
}: ColorUtil) => theme.colorCompliment[color] || defaultColor;

export const colorOrShade = (shade: $Keys<ColorShade> = "lightGray") => ({
  color,
  theme
}: ColorUtil) => theme.color[color] || themeColorShade(shade)({ theme });

export const complimentOrShade = (shade: $Keys<ColorShade> = "lightGray") => ({
  color,
  theme
}: ColorUtil) =>
  theme.colorCompliment[color] || themeColorShade(shade)({ theme });
