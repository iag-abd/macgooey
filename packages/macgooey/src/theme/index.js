// @flow

export { ThemeProvider } from "styled-components";

export type ColorPalette = {|
  primary: string,
  secondary: string,
  tertiary: string
|};

export type ColorShade = {|
  lighterGray: string,
  lightGray: string,
  gray: string,
  darkGray: string,
  darkerGray: string
|};

type ElementStyles = {|
  borderRadius: string,
  borderWidth: string
|};

type ButtonStyles = ElementStyles;

type InputStyles = ElementStyles;

export type Styles = {|
  button: ButtonStyles,
  color: ColorPalette,
  colorCompliment: ColorPalette,
  colorShade: ColorShade,
  fontFamilyHeading: string,
  input: InputStyles,
  textColorBody: string
|};

export const themeBase: Styles = {
  fontFamilyHeading: "'Open Sans', sans-serif",
  textColorBody: "#333",

  button: {
    borderRadius: "3px",
    borderWidth: "0"
  },

  input: {
    borderWidth: "1px",
    borderRadius: "0"
  },

  color: {
    primary: "#4375E8",
    secondary: "#FF4448",
    tertiary: "#E8B221"
  },
  colorCompliment: {
    primary: "#FFF",
    secondary: "#FFF",
    tertiary: "#FFF"
  },
  colorShade: {
    lighterGray: "#eee",
    lightGray: "#ddd",
    gray: "#aaa",
    darkGray: "#555",
    darkerGray: "#222"
  }
};

export const themeAerith = {
  ...themeBase,

  fontFamilyHeading: "'Montserrat', sans-serif",
  textColorBody: "#414042",

  input: {
    borderWidth: "1px",
    borderRadius: "0"
  },

  color: {
    default: "#DDD",
    primary: "#0BAB47",
    secondary: "#6D6E71",
    tertiary: "#007BAD"
  },
  colorCompliment: {
    default: "#333",
    primary: "#FFF",
    secondary: "#FFF",
    tertiary: "#FFF"
  }
};

export const themeSephiroth = {
  ...themeBase,

  fontFamilyHeading: "'Source Sans Pro', 'Open Sans', sans-serif",
  textColorBody: "#3B3D40",

  button: {
    borderRadius: "3px",
    borderWidth: "1px"
  },

  color: {
    default: "#F0F0F0",
    primary: "#602f89",
    secondary: "#01B5CC",
    tertiary: "#1E2741"
  },
  colorCompliment: {
    default: "#602F89",
    primary: "#FFF",
    secondary: "#FFF",
    tertiary: "#FFF"
  }
};
