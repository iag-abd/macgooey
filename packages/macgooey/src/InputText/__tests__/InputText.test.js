// @flow

import React from "react";
import { InputText } from "..";
import { themeBase as theme } from "../../theme";
import "jest-styled-components";

import renderer from "react-test-renderer";

describe("<InputText />", () => {
  test("it should render", () => {
    const component = renderer.create(<InputText theme={theme} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("<InputText placeholder />", () => {
  test("it should render", () => {
    const component = renderer.create(
      <InputText placeholder="foo" theme={theme} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
