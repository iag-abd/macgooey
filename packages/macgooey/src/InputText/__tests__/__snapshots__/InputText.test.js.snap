// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText /> it should render 1`] = `
.c0 {
  padding: 10px;
  font-family: 'Open Sans',sans-serif;
  font-size: 1em;
  border-width: 1px;
  border-color: #ddd;
  border-style: solid;
  outline: none;
  width: 100%;
  -webkit-transition: border-color 150ms;
  transition: border-color 150ms;
}

.c0:focus {
  border-color: #4375E8;
}

.c0::-webkit-input-placeholder {
  color: #aaa;
}

.c0::-moz-placeholder {
  color: #aaa;
}

.c0:-ms-input-placeholder {
  color: #aaa;
}

.c0::placeholder {
  color: #aaa;
}

<input
  className="c0"
/>
`;

exports[`<InputText placeholder /> it should render 1`] = `
.c0 {
  padding: 10px;
  font-family: 'Open Sans',sans-serif;
  font-size: 1em;
  border-width: 1px;
  border-color: #ddd;
  border-style: solid;
  outline: none;
  width: 100%;
  -webkit-transition: border-color 150ms;
  transition: border-color 150ms;
}

.c0:focus {
  border-color: #4375E8;
}

.c0::-webkit-input-placeholder {
  color: #aaa;
}

.c0::-moz-placeholder {
  color: #aaa;
}

.c0:-ms-input-placeholder {
  color: #aaa;
}

.c0::placeholder {
  color: #aaa;
}

<input
  className="c0"
  placeholder="foo"
/>
`;
